diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index 56cd33d..071132e 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -133,7 +133,6 @@ PRODUCT_PACKAGES += \
 
 # Optional CM packages
 PRODUCT_PACKAGES += \
-    VideoEditor \
     VoiceDialer \
     SoundRecorder \
     Basic
@@ -146,7 +145,6 @@ PRODUCT_PACKAGES += \
     audio_effects.conf \
-    CMWallpapers \
     Apollo \
     CMUpdater \
     CMFileManager
 
 # Extra tools in CM
diff --git a/vendor/cm/config/common_full.mk b/vendor/cm/config/common_full.mk
index 8d69ead..342b1c7 100644
--- a/vendor/cm/config/common_full.mk
+++ b/vendor/cm/config/common_full.mk
@@ -8,7 +8,7 @@ include frameworks/base/data/sounds/NewAudio.mk
 include frameworks/base/data/sounds/AudioPackageNewWave.mk
 
 # Bring in all video files
-$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
+#$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
 
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk
