diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
--- a/vendor/cm/config/common.mk	2013-06-09 12:41:27.719460114 +0600
+++ b/vendor/cm/config/common.mk	2013-06-10 15:58:06.277579651 +0600
@@ -88,7 +88,7 @@
 
 # Optional CM packages
 PRODUCT_PACKAGES += \
-    VideoEditor \
+#    VideoEditor \
     VoiceDialer \
     SoundRecorder \
     Basic \
@@ -107,7 +107,7 @@
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
-    CMWallpapers \
+#    CMWallpapers \
     Apollo \
     CMUpdater
 
diff --git a/vendor/cm/config/common_full_phone.mk b/vendor/cm/config/common_full_phone.mk
index 8d69ead..342b1c7 100644
--- a/vendor/cm/config/common_full_phone.mk
+++ b/vendor/cm/config/common_full_phone.mk
@@ -8,7 +8,7 @@ include frameworks/base/data/sounds/NewAudio.mk
 include frameworks/base/data/sounds/AudioPackageNewWave.mk
 
 # Bring in all video files
-$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
+#$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
 
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk

