diff --git a/frameworks/base/services/surfaceflinger/SurfaceFlinger.cpp b/frameworks/base/services/surfaceflinger/SurfaceFlinger.cpp
--- a/frameworks/base/services/surfaceflinger/SurfaceFlinger.cpp	2013-06-09 12:41:27.719460114 +0600
+++ b/frameworks/base/services/surfaceflinger/SurfaceFlinger.cpp	2013-06-10 15:58:06.277579651 +0600
@@ -872,7 +872,8 @@
     mSwapRegion.orSelf(mDirtyRegion);
 
     if (UNLIKELY(mDebugRegion)) {
-        debugFlashRegions();
+	//disable debugFlashRegions to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+        //debugFlashRegions();
     }
 
     // set the frame buffer
@@ -913,10 +914,11 @@
 
     setupHardwareComposer(mDirtyRegion);
 #ifdef QCOM_HARDWARE
-    if (!mCanSkipComposition)
-        composeSurfaces(mDirtyRegion);
+	//disable composeSurfaces to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+    //if (!mCanSkipComposition)
+        //composeSurfaces(mDirtyRegion);
 #else
-    composeSurfaces(mDirtyRegion);
+    //composeSurfaces(mDirtyRegion);
 #endif
 
     // update the swap region and clear the dirty region
@@ -1081,14 +1083,17 @@
             if (!region.isEmpty()) {
                 if (0 != qcomuiClearRegion(region, hw.getEGLDisplay(),
                                              hw.getEGLSurface())) {
-                    drawWormhole();
+			//disable drawWormhole to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+                    //drawWormhole();
                 }
             }
         } else {
-            drawWormhole();
+		//disable drawWormhole to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+            //drawWormhole();
         }
 #else
-        drawWormhole();
+	//disable drawWormhole to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+        //drawWormhole();
 #endif
     }
 
@@ -2321,7 +2326,8 @@
     invalidateHwcGeometry();
 
     if (mode & ISurfaceComposer::eElectronBeamAnimationOff) {
-        electronBeamOffAnimationImplLocked();
+	//disable screenOnAnimation to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+        //electronBeamOffAnimationImplLocked();
     }
 
     // always clear the whole screen at the end of the animation
@@ -2377,7 +2383,8 @@
         return NO_ERROR;
     }
     if (mode & ISurfaceComposer::eElectronBeamAnimationOn) {
-        electronBeamOnAnimationImplLocked();
+	//disable screenOnAnimation to prevent visual bugs with 240x400 screen were scaled by mdp in kernel
+        //electronBeamOnAnimationImplLocked();
     }
 
     // make sure to redraw the whole screen when the animation is done
